README.md

```diff
@@ -86,46 +87,87 @@ Ensure you have the following installed:
-    # build the container
+    # pull the supporting containers
+    make compose-pull
+
+    # build the main application image
     make docker-build
-    # start the container
-    make docker-start
+
+    # start the containers
+    make compose-up
     ```
 
 2. **Migrations and Superuser Creation**
 
     If this is your first time setting up the project, you’ll need to run migrations to set up the database schema and create an administrative user.
-    Note: `dj` is an alias for `django-admin`
 
     ```bash
     # ssh into the web container
     make docker-shell
 
     # Run database migrations
-    dj migrate
+    make migrate
 
     # Create a superuser for accessing the admin interface
-    dj createsuperuser
+    make createsuperuser
     ```
 
 3. **Start Django Inside the Container**
 
     Once the containers are running, you need to manually start Django from within the web container. This allows for running both the Django server and any additional background services (e.g., schedulers).
 
+> [!WARNING]  
+> The `honcho` command will pick up your local mounted `.env` file when running via `docker-compose`. Ensure that you comment out any local variables which might cause clashes in the container context as they will take precedence when running `honcho start`.
+
     ```bash
     # Start both Django and the scheduler using Honcho
     honcho start
 
-    # To run just the web server. alias: djrun
     # This is not needed if you used `honcho start`
-    dj runserver 0.0.0.0:8000
+    make runserver
     ```
 
 You can then access the admin at `http://0.0.0.0:8000/admin/` or `http://localhost:8000/admin/`.
 
+#### Running locally with supporting services in Docker
+
+You can also run the main application locally with the supporting backend services such as the Postgres and Redis running in Docker. This can be useful when you want to make changes that require the app to be frequently restarted to be picked up.
+
+In order to run it:
+
+1. Pull the images of the supporting services.
+
+```
+make compose-dev-pull
+```
+
+2. Start the supporting services in Docker.
+
+```
+make compose-dev-up
+```
+
+3. Run the below command to run the necessary pre-run steps:
+
+```
+make dev-init
+```
+
+4. Run the Django server locally via your IDE or with the following command:
+
+```
+make runserver
+```
+
+You can specify the runtime configuration either in your IDE (for PyCharm see [here](https://www.jetbrains.com/help/pycharm/run-debug-configuration.html#createExplicitly)), or copy the `.env-dev-example` and rename it to `.env` which will allow Django to pick up the config.
+Note that once you create the `.env` file, and you'd like to switch back to running the application in a container with `make compose-up`, the `.env` file will be accessible inside the containers and it will be picked up by the `honcho` command. In order to avoid conflicts you should comment out the `DATABASE_URL` and `REDIS_URL` variables in the `.env` file.
```

Comment:
```suggestion
Note that once you create the `.env` file, and you'd like to switch back to running the application in a container with `make compose-up`, the `.env` file will be accessible inside the containers and it will be picked up by the `honcho` command. In order to avoid conflicts you may need to comment out some variables (such as `DATABASE_URL` and `REDIS_URL`).
```